File: day09-链表/203.移出链表的元素/203.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function (head, val) {
  let tem = new ListNode(0, head);
  let res = tem;
  while (tem.next) {
    if (tem.next.val === val) {
      tem.next = tem.next.next;
      continue;
    }
    tem = tep.next;
  }
  return res.next;
};

File: day09-链表/203.移出链表的元素/203.java
/**
 * Definition for singly-linked list. public class ListNode { int val; ListNode
 * next; ListNode() {} ListNode(int val) { this.val = val; } ListNode(int val,
 * ListNode next) { this.val = val; this.next = next; } }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode res = new ListNode(0, head);
        ListNode res2 = res;
        while (res.next != null) {
            if (res.next.val == val) {
                res.next = res.next.next;
                continue;
            }
            res = res.next;
        }
        return res2.next;
    }
}